src/Schemes/AffineSchemes/Morphisms/Constructors.jl

```diff
@@ -0,0 +1,182 @@
+
+@Markdown.doc """
+    SpecMor(X::AbsSpec, Y::AbsSpec, f::Vector{<:RingElem}; check::Bool=true)
+
+This method constructs a morphism from the scheme ``X``
+to the scheme ``Y``. For this one has to specify the images
+of the coordinates (the generators of `ambient_ring(Y)`)
+under the pullback map ``𝒪(Y) → 𝒪(X)`` as third argument.
+
+Note that expensive checks can be turned off by setting `check=false`.
+
+# Examples
+```jldoctest
+julia> X = affine_space(QQ,3)
+Spec of Multivariate Polynomial Ring in x1, x2, x3 over Rational Field
+
+julia> Y = affine_space(QQ,3)
+Spec of Multivariate Polynomial Ring in x1, x2, x3 over Rational Field
+
+julia> SpecMor(X, Y, gens(OO(X)));
+```
+"""
+function SpecMor(
+      X::AbsSpec,
+      Y::AbsSpec,
+      f::Vector{<:RingElem};
+      check::Bool=true
+  )
+  return SpecMor(X, Y, hom(OO(Y), OO(X), OO(X).(f), check=check), check=check)
+end
+
+function SpecMor(
+      X::AbsSpec,
+      Y::AbsSpec{<:Ring, <:MPolyRing},
+      f::Vector{<:RingElem};
+      check::Bool=true
+  )
+  return SpecMor(X, Y, hom(OO(Y), OO(X), OO(X).(f)), check=check)
+end
+
+function SpecMor(
+      X::AbsSpec,
+      Y::AbsSpec,
+      f::Vector;
+      check::Bool=true
+  )
+  return SpecMor(X, Y, OO(X).(f), check=check)
+end
+
+
+
+########################################################################
+# (2) Special constructors
+########################################################################
+
+@Markdown.doc """
+    identity_map(X::AbsSpec{<:Any, <:MPolyRing})
+
+This method constructs the identity morphism from an affine scheme to itself.
+
+# Examples
+```jldoctest
+julia> X = affine_space(QQ,3)
+Spec of Multivariate Polynomial Ring in x1, x2, x3 over Rational Field
+
+julia> identity_map(X);
+```
+"""
+identity_map(X::AbsSpec{<:Any, <:MPolyRing}) = SpecMor(X, X, hom(OO(X), OO(X), gens(OO(X))))
+identity_map(X::AbsSpec{<:Any, <:MPolyQuoLocalizedRing}) = SpecMor(X, X, hom(OO(X), OO(X), gens(ambient_ring(X)), check=false))
+identity_map(X::AbsSpec{<:Any, <:MPolyLocalizedRing}) = SpecMor(X, X, hom(OO(X), OO(X), gens(ambient_ring(X)), check=false))
+identity_map(X::AbsSpec{<:Any, <:MPolyQuo}) = SpecMor(X, X, hom(OO(X), OO(X), gens(ambient_ring(X))))
+
+
+@Markdown.doc """
+    inclusion_map(X::AbsSpec, Y::AbsSpec)
+
+This method constructs the inclusion map from ``X`` to ``Y``.
+For convenience, also the method `inclusion_morphism` is supported.
+
+# Examples
+```jldoctest
+julia> X = affine_space(QQ,3)
+Spec of Multivariate Polynomial Ring in x1, x2, x3 over Rational Field
+
+julia> R = OO(X)
+Multivariate Polynomial Ring in x1, x2, x3 over Rational Field
+
+julia> (x1,x2,x3) = gens(R)
+3-element Vector{fmpq_mpoly}:
+ x1
+ x2
+ x3
+
+julia> Y = subscheme(X, x1)
+Spec of Quotient of Multivariate Polynomial Ring in x1, x2, x3 over Rational Field by ideal(x1)
+
+julia> inclusion_map(Y, X);
+```
+"""
+inclusion_map(X::AbsSpec, Y::AbsSpec) = SpecMor(X, Y, gens(ambient_ring(Y)))  # TODO: Remove
```

Comment:
Where is the problem about removing this line? Is it used somewhere outside this pull request?

Reply:
I was hoping we could have this PR solemnly for moving the existing code and documenting it. Improving/streamlining it should imho be a separate PR (not only because this PR is already massive as it is). This is why I did not yet touch this/try code optimizations.

Reply:
But it seems we are trying both now... so let me just do it.